src/engine/Match_extract_mode.ml

```diff
@@ -214,8 +214,8 @@ let map_taint_trace map_loc { Pattern_match.source; tokens; sink } =
           }
   in
   {
-    Pattern_match.source = map_taint_call_trace source;
-    tokens = Common.map map_loc tokens;
+    Pattern_match.sources =
+      Common.map (fun (ct, toks) -> (map_taint_call_trace ct, toks)) sources;
```

Comment:
You are removing the `map_loc` that was applied here to `tokens`, is it on purpose?

Reply:
nope, good catch